llvm/lib/Target/AMDGPU/AMDGPUUnifyDivergentExitNodes.cpp

```diff
@@ -215,7 +215,10 @@ bool AMDGPUUnifyDivergentExitNodesImpl::run(Function &F, DominatorTree *DT,
       PDT.roots(), [&](auto BB) { return !isUniformlyReached(UA, *BB); });
 
   for (BasicBlock *BB : PDT.roots()) {
-    if (isa<ReturnInst>(BB->getTerminator())) {
+    if (auto *RI = dyn_cast<ReturnInst>(BB->getTerminator())) {
+      auto *CI = dyn_cast_or_null<CallInst>(RI->getPrevNode());
```

Comment:
Is it enforced this must be the exact previous instruction?

Reply:
[Yes](https://llvm.org/docs/LangRef.html#call-instruction)